### PR TITLE
Move InMemorySubscriber from L2 communication module to  L3 client mo…

### PR DESCRIPTION
**Pull request by stevenhartley**
The InMemorySubscriber actually was implementing the client-side of the uSubscription flow by talking to usubscription service, registering a listener (to receive published messages), and setting up a notifier to receive subscription changes. The only L2 item for pub/sub (subscriber) flow was to register a listener with the transport so there is no need to add a wrapper to do that. This change then also implements all the other remaining usubscription client side APIs for developers to use and removes the L2 Subscriber to avoid confusion with this InMemoryUSubscriptionClient implementation. The uProtocol client-side implementations will now reside in the client folder of up-java (ex. uDiscovery & uTwin).

#148

### CODE REVIEW

File: src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java [hunk 1]
```diff
@@ -39,45 +41,84 @@
...
+     * Creates a new USubscription client passing {@link UTransport}, {@link CallOptions},
+     * and an implementation of {@link RpcClient} and {@link Notifier}.
+     * 
+     * @param transport the transport to use for sending the notifications
+     * @param rpcClient the rpc client to use for sending the RPC requests
+     * @param notifier the notifier to use for registering the notification listener
+     * @param options the call options to use for the RPC requests
+     */
+    public InMemoryUSubscriptionClient (UTransport transport, RpcClient rpcClient,
+        Notifier notifier, CallOptions options) {
```
Comment (mishap4) [inline src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java:117]: CallOptions contains auth token and it is immutable. When token expires it has to be refreshed, so the current implementation will require to rebuild the whole instance just to update token, I think it is better to pass options into each RPC method.

Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java:117]: ok fixed

**Review by mishap4 — APPROVED** [review]

File: src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java
[the remark this answers is not in the document]
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java:1]: done

[the remark this answers is not in the document]
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java:1]: done

File: src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java [hunk 2]
```diff
@@ -98,8 +115,8 @@ public InMemorySubscriber (UTransport transport, RpcClient rpcClient, Notifier n
      * subscribed to said topic. 
      * 
      * @param topic The topic to subscribe to.
-     * @param listener The listener to be called when a message is received on the topic.
-     * @param options The call options for the subscription.
+     * @param listener The listener to be called when a messages are received.
```
[the remark this answers is not in the document]
Comment (stevenhartley) [reply src/main/java/org/eclipse/uprotocol/client/usubscription/v3/InMemoryUSubscriptionClient.java]: done

**Review by neelam-kushwah — APPROVED** [review]
LGTM

**Pull request merged**